docs/proposal-0008-daily_metric_aggregations.md

```diff
@@ -0,0 +1,133 @@
+  * **Con**: Significant increase in number of queries & tables
+  * **Con**: Increased complexity in debugging (e.g., ~10x number of tasks in Argo UI)
+  * This capability is largely built but kept behind a CLI flag
+    * Ideally we can use both capabilities, for different scenarios:
+      * Daily Aggregations for all experiments
+      * Discrete metrics for reruns and case-by-case (and using daily aggregations)
+
+
+## Changes Required
+
+This solution requires some significant changes to how Jetstream currently works. I will describe the relevant parts of the workflow, and where it would need to change to accommodate the proposed behavior.
+
+#### Current
+1. Jetstream reads experiment config, grouping metrics into their respective analysis windows.
+2. Jetstream loops over the analysis windows. For each window:
+    - Ensure this window is in the list to be computed (CLI parameter)
+    - Ensure this window should be computed for the current execution date (i.e., daily computed every day, weekly computed on `mod-7 + 1` days, overall computed at end)
+    - For each analysis basis
+      - Build and execute metrics query
+      - Get metric results
+      - Compute statistics for each metric
+      - Write statistics to table
+    - Produce metrics and statistics views containing results from all analysis windows within a given period
+Note: all tasks in (2) are spawned in parallel, managed by dask, so the loops do not get blocked waiting for results
+
+#### Proposed
+1. Jetstream reads experiment config, **grouping metrics into their respective analysis windows** (optional).
+2. Jetstream loops over the analysis windows. For **daily** window:
+    - **change** start by checking whether table exists and returning early if yes
+      - these tables may get deleted by a rerun, but that happens upfront and so checking for existence of the table should be reasonable to do here
+    - After check, same as current workflow -- see caveat below for possible change
+2. (**changes**) For **other** windows:
+    - Same checks as above to ensure we should continue with execution.
+    - Each task should have a dependence on the current daily execution(s), as well as all other relevant daily windows (e.g., if running week 0, day 6 should be already kicked off, and days 0-5 should be added as dependency here as well)
+      - if day's results already exist, it will return early (see change to daily window above)
+    - If all daily results exist, query daily metrics view for relevant window indices
+      - Aggregate per-client metric values based on the metric definition [Question 1]
+
+
+#### Questions
+1. How to handle aggregating client metric values across multiple days?
+    - Query could look like the following
+      - `MIN` enrollment and exposure dates is not necessary since we already do this when building enrollments
+      - `SUM` enrollment and exposure events
+      - `LOGICAL_OR` for boolean metrics
+      - `SUM` for int metrics
```

Comment:
I think this section needs to be more fleshed out. IMO, this is really the meat of the proposal so we should align on what options we'd consider, what their pros/cons are, etc. before we can make a decision.